app_architecture/ruby/ruby_on_rails.adoc

```diff
@@ -0,0 +1,332 @@
+----
+
+Then move into your new application directory.
+
+----
+$ cd rails-app
+----
+
+In case you already have an application, make sure the `pg` gem is present in your `Gemfile`. If not
+edit your `Gemfile` by adding the gem:
+
+----
+gem 'pg'
+----
+
+To generate a new `Gemfile.lock` with all your dependencies run:
+
+----
+$ bundle install
+----
+
+In addition to using the `postgresql` database with `pg` gem, you'll also need to ensure the `config/databese.yml` is using the `postgresql` adapter.
+
+Make sure your development section of your `config/database.yml` file looks like this:
+
+----
+# PostgreSQL. Versions 8.2 and up are supported.
+#
+# Install the pg driver:
+#   gem install pg
+# On OS X with Homebrew:
+#   gem install pg -- --with-pg-config=/usr/local/bin/pg_config
+# On OS X with MacPorts:
+#   gem install pg -- --with-pg-config=/opt/local/lib/postgresql84/bin/pg_config
+# On Windows:
+#   gem install pg
+#       Choose the win32 build.
+#       Install PostgreSQL and put its /bin directory on your path.
+#
+# Configure Using Gemfile
+# gem 'pg'
+#
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  # For details on connection pooling, see rails configuration guide
+  # http://guides.rubyonrails.org/configuring.html#database-pooling
+  pool: 5
+----
+
+[WARNING]
+====
+*ToBeDone*:
+We in OpenShift need a way how to complete the database.yml based on the detected available environment
+variables, Gemfile gems and set adapter in `config/database.yml`. For example how the final file should look like, check out our link:https://github.com/openshift/rails-ex[Ruby on Rails example application] link:https://github.com/openshift/rails-ex/blob/master/config/database.yml[config/database.yml].
+====
+
+[[rails-welcome-page]]
+
+=== Welcome page
+
+Since Rails 4 no longer servers static `public/index.html` page in production, we need to create a 
+new root page.
+
+In order to have a custom welcome page we need to do following steps:
+
+* Create a *controller* with an index action 
+* Create a *view* page for the `welcome` controller `index` action
+* Create a *route* that will serve applications root page with the created *controller* and *view*  
+
+Rails offers a generator that will do all this necessary steps for you. Just type:
+
+----
+$ rails generate controller welcome index
+----
+
+All the necessary files have been created, now we just need to edit line 2 in `config/routes.rb`
+file to look like:
+----
+root 'welcome#index'
+---- 
+
+Run the rails server to verify the page is available
+
+----
+$ rails server
+----
+
+You should see your page by visiting link:http://localhost:3000[http://localhost:3000] in our browser.
+I case you don't see the page, check the link:TBD[logs] that are output to your server to debug.
+
+[[store-your-app-in-git]]
+
+=== Store your App in Git
+
+OpenShift relies on link:http://git-scm.com/[git] , a distributed source control management tool, for
```

Comment:
trim this down.   "OpenShift requires git, if you don't have it installed you will need to install it."

strip out the "run git --help", etc.    We want this to look simple :)